src/node.h

```diff
@@ -1031,6 +1031,13 @@ v8::MaybeLocal<v8::Value> MakeCallback(v8::Isolate* isolate,
                                        v8::Local<v8::Value>* argv,
                                        async_context asyncContext);
 
+NODE_EXTERN
+v8::MaybeLocal<v8::Value> MakeSyncCallback(v8::Isolate* isolate,
+                                           v8::Local<v8::Object> recv,
+                                           v8::Local<v8::Function> callback,
+                                           int argc,
+                                           v8::Local<v8::Value> argv[]);
```

Comment:
I would rather not start exposing an public API like this for the needs of our internals … if you disagree, that’s okay, but in that case it should come with explicit documentation about the use cases and limitations of doing this

Reply:
> I would rather not start exposing an public API like this for the needs of our internals … if you disagree, that’s okay, but in that case it should come with explicit documentation about the use cases and limitations of doing this

No objections to that, of course. We only figured that keeping the MakeSyncCallback() implementation close to the existing MakeCallback() family would help maintainability in the future, hence we chose to put it into callback.cc (rather than e.g. node_perf.cc). But we can very well make it internal. Just a question: would moving the declaration to node_internals.h and dropping NODE_EXTERN suffice for that?

Reply:
> Just a question: would moving the declaration to node_internals.h and dropping NODE_EXTERN suffice for that?

Yeah, that should be fine :+1:

Reply:
> > Just a question: would moving the declaration to node_internals.h and dropping NODE_EXTERN suffice for that?
> 
> Yeah, that should be fine +1

Many thanks! Done.